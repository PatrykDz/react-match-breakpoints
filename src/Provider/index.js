import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Context from '../Context'

export let matchMediaBreakpoints = {}

class Provider extends Component {
    state = {
        breakpoints: {},
        componentRenameFn: undefined
    }
    constructor (props) {
        super(props)
        const { breakpoints, componentRenameFn } = props

        matchMediaBreakpoints = this.buildMatchMediaBreakpoints(breakpoints, componentRenameFn)

        this.state = {
            ...Object.keys(matchMediaBreakpoints).reduce((acc, breakpoint) => {
                acc[breakpoint] = matchMediaBreakpoints[breakpoint].matches
                return acc
            }, {}),
        }
    }Å›

    buildMatchMediaBreakpoints = () => {

    }

    addBreakpointsListeners = () => {
        Object.keys(matchMediaBreakpoints).forEach(breakpoint => {
            matchMediaBreakpoints[breakpoint].addListener(mq => {
                mq.matches ? this.setActiveBreakpoint(breakpoint) : this.unsetActiveBreakpoint(breakpoint)
            })
        })
    }

    setActiveBreakpoint = breakpoint => {
        this.setState({
            [breakpoint]: true
        })
    }

    unsetActiveBreakpoint = breakpoint => {
        this.setState({
            [breakpoint]: false
        })
    }

    render() {
        return (
            <Context.Provider value={this.state}>
                {this.props.children}
            </Context.Provider>
        )
    }
}

Provider.propTypes = {
    breakpoints: PropTypes.object.isRequired,
}

export default Provider