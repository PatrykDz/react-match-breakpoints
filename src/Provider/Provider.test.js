import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Provider from '../Provider'

const breakpoints = {
  isMobile: `screen and (max-width: 500px)`,
  isTablet: `screen and (min-width: 500px) and (max-width: 1200px)`,
  isDesktop: `screen and (min-width: 1201px)`,
}

const matchMediaBreakpoints = {
  isMobile: window.matchMedia(`screen and (max-width: 500px)`),
  isTablet: window.matchMedia(`screen and (min-width: 500px) and (max-width: 1200px)`),
  isDesktop: window.matchMedia(`screen and (min-width: 1201px)`),
}

const stateMediaBreakpoints = {
  isMobile: false,
  isTablet: false,
  isDesktop: false,
}

beforeEach(() => {
  jest.resetModules()
})

describe('<Provider />', () => {
  it('should render Provider', () => {
    shallow(
      <Provider breakpoints={breakpoints}>
        <div>test</div>
      </Provider>
    )
  })

  it('matches the snapshot', () => {
    const wrapper = shallow(
      <Provider breakpoints={breakpoints}>
        <div>test</div>
      </Provider>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

describe('<Provider /> buildBooleanBreakpoints method', () => {
  it('should return object with boolean values for breakpoints when provided mq values', () => {
    const wrapper = shallow(
      <Provider breakpoints={breakpoints}>
        <div>test</div>
      </Provider>
    )
    const returnObject = wrapper.instance().buildBooleanBreakpoints(matchMediaBreakpoints)
    expect(returnObject).toEqual(stateMediaBreakpoints)
  })
})
