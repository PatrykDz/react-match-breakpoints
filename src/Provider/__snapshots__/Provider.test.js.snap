// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Provider /> matches the snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "isDesktop": false,
      "isMobile": false,
      "isTablet": false,
    }
  }
>
  <div>
    test
  </div>
</ContextProvider>
`;
